perun/collect/complexity/configurator.py

```diff
@@ -97,14 +102,23 @@ def _config_write_config(config_handle, executable_path, runtime_filter, include
         sample_map = _config_create_sample(include_list, config['sampling'])
     filter_list, sample_dict = _config_symbols_to_addresses(executable_path, runtime_filter, sample_map)
 
-    # Append the file name configuration
-    conf = {'file-name': config['file-name']}
-    # Append the storage size configuration
-    if 'init-storage-size' in config:
-        conf['init-storage-size'] = config['init-storage-size']
+    # Handle the possibly missing internal configuration details
+    if 'internal_data_filename' not in config:
+        config['internal_data_filename'] = DEFAULT_DATA_FILENAME
+    if 'internal_storage_size' not in config:
+        config['internal_storage_size'] = DEFAULT_STORAGE_SIZE
+    if 'internal_direct_output' not in config:
+        config['internal_direct_output'] = DEFAULT_DIRECT_OUTPUT
+
+    # Append the internal configuration
+    conf = {
+        'internal_data_filename': config['internal_data_filename'],
```

Comment:
Updated the dict initialization code and changed the identifiers to better reflect the meaning of both dictionaries. The dictionary passed to the method contains more keys (collector job data) and thus new configuration dictionary is created (the internal collector configuration with less keys).